lib/ImageResize.php

```diff
@@ -7,316 +7,333 @@
 /**
  * PHP class to resize and scale images
  */
-class ImageResize
+class ImageResize implements ImageResizeInterface
 {
-    const CROPTOP = 1;
-    const CROPCENTRE = 2;
-    const CROPCENTER = 2;
-    const CROPBOTTOM = 3;
-    const CROPLEFT = 4;
-    const CROPRIGHT = 5;
-    const CROPTOPCENTER = 6;
-    const IMG_FLIP_HORIZONTAL = 0;
-    const IMG_FLIP_VERTICAL = 1;
-    const IMG_FLIP_BOTH = 2;
-
+    /**
+     * @var int JPEG output quality
+     */
     public $quality_jpg = 85;
+    
+    /**
+     * @var int WEBP output quality
+     */
     public $quality_webp = 85;
+    
+    /**
+     * @var int PNG output quality
+     */
     public $quality_png = 6;
+    
+    /**
+     * @var bool
+     */
     public $quality_truecolor = true;
-    public $gamma_correct = true;
-
+    
+    /**
+     * @var bool
+     */
+    public $gamma_correction = false;
+    
+    /**
+     * @var int
+     */
     public $interlace = 1;
-
+    
+    /**
+     * @var mixed
+     */
     public $source_type;
-
+    
     protected $source_image;
-
+    
     protected $original_w;
     protected $original_h;
-
+    
     protected $dest_x = 0;
     protected $dest_y = 0;
-
+    
     protected $source_x;
     protected $source_y;
-
+    
     protected $dest_w;
     protected $dest_h;
-
+    
     protected $source_w;
     protected $source_h;
-
+    
     protected $source_info;
-
+    
     protected $filters = [];
-
-    /**
-     * Create instance from a strng
-     *
-     * @param string $image_data
-     * @return ImageResize
-     * @throws ImageResizeException
-     */
+    
     public static function createFromString($image_data)
     {
-        if (empty($image_data) || $image_data === null) {
-            throw new ImageResizeException('image_data must not be empty');
+        if (empty( $image_data ) || $image_data === null) {
+            throw new ImageResizeException( __CLASS__ . ' ERROR: image_data must not be empty.' );
         }
-        $resize = new self('data://application/octet-stream;base64,' . base64_encode($image_data));
-        return $resize;
+        return new self( 'data://application/octet-stream;base64,'.base64_encode( $image_data ) );
```

Comment:
Please consider reverting the previous coding style.

Reply:
PHP compiler will optimize this code without any problems

Reply:
I can't understand you. You need in `$resize` local value?